lizmap/modules/lizmap/lib/App/XmlTools.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * app context informations inside a Jelix environment.
+ *
+ * @author    3liz
+ * @copyright 2020 3liz
+ *
+ * @see      https://3liz.com
+ *
+ * @license Mozilla Public License : http://www.mozilla.org/MPL/
+ */
+
+namespace Lizmap\App;
+
+class XmlTools
+{
+    /*
+     * Interprets a string of XML into an object
+     *
+     * @param $xml_str a well-formed XML string
+     *
+     * @return ($xml, $msg) Array containing $xml an object of class SimpleXMLElement with properties
+     * containing the data held within the XML document, or false on failure ; and $msg a string
+     * containing the error message.
+     */
+    public static function xmlFromString($xml_str)
+    {
+        $use_errors = libxml_use_internal_errors(true);
+        $xml = simplexml_load_string($xml_str);
+        $msg = '';
+        if (!$xml) {
+            $msg = self::xmlErrorMsg();
+        }
+
+        return array($xml, $msg);
+    }
+
+    /*
+     * Interprets an XML file into an object
```

Comment:
same issue in `@param` and `@return` syntax. See https://docs.phpdoc.org/latest/guide/guides/types.html